cadence/contracts/DapperStorageRent.cdc

```diff
@@ -0,0 +1,247 @@
+  /// List of all refilledAccounts
+  access(contract) var RefilledAccounts: [Address]
+  /// Detailed account information of refilled accounts
+  access(contract) var RefilledAccountInfos: {Address: RefilledAccountInfo}
+  /// List of all blockedAccounts
+  access(contract) var BlockedAccounts: [Address]
+  /// Blocks required between refill attempts
+  access(contract) var RefillRequiredBlocks: UInt64
+
+  /// Event emitted when an Admin blocks an address
+  pub event BlockedAddress(_ address: [Address])
+  /// Event emitted when a Refill is successful
+  pub event Refuelled(_ address: Address)
+  /// Event emitted when a Refill is not successful
+  pub event RefilledFailed(address: Address, reason: String)
+
+  /// getStorageRentRefillThreshold
+  /// Get the current StorageRentRefillThreshold
+  ///
+  /// @return UInt64 value of the current StorageRentRefillThreshold value
+  pub fun getStorageRentRefillThreshold(): UInt64 {
+    return self.StorageRentRefillThreshold
+  }
+
+  /// getRefilledAccounts
+  /// Get the current StorageRentRefillThreshold
+  ///
+  /// @return List of refilled Accounts
+  pub fun getRefilledAccounts(): [Address] {
+    return self.RefilledAccounts
+  }
+
+  /// getBlockedAccounts
+  /// Get the current StorageRentRefillThreshold
+  ///
+  /// @return List of blocked accounts
+  pub fun getBlockedAccounts() : [Address] {
+    return self.BlockedAccounts
+  }
+
+  /// getRefilledAccountInfos
+  /// Get the current StorageRentRefillThreshold
+  ///
+  /// @return Address: RefilledAccountInfo mapping
+  pub fun getRefilledAccountInfos(): {Address: RefilledAccountInfo} {
+    return self.RefilledAccountInfos
+  }
+
+  /// getRefillRequiredBlocks
+  /// Get the current StorageRentRefillThreshold
+  ///
+  /// @return UInt64 value of the current RefillRequiredBlocks value
+  pub fun getRefillRequiredBlocks(): UInt64 {
+    return self.RefillRequiredBlocks
+  }
+
+  /// tryRefill
+  /// Attempt to refill an accounts storage capacity if it has dipped below threshold and passes other checks.
+  ///
+  /// @param address: Address to attempt a storage refill on
+  pub fun tryRefill(_ address: Address) {
+    let REFUEL_AMOUNT = 0.06;
+
+    self.cleanExpiredRefilledAccounts(10)
+
+    // Get the Flow Token reciever of the address
+    let recipient = getAccount(address)
+    let receiverRef = recipient.getCapability<&PrivateReceiverForwarder.Forwarder>(PrivateReceiverForwarder.PrivateReceiverPublicPath).borrow()
+
+    // Sliently fail if the receiverRef is nill
+    if receiverRef == nil || receiverRef!.owner == nil {
+      emit RefilledFailed(address: address, reason: "Couldn't borrow the Accounts flowTokenVault")
+      return
+    }
+
+    // Sliently fail if the account has already be refueled within the block allowance
+    if self.RefilledAccountInfos[address] != nil && getCurrentBlock().height - self.RefilledAccountInfos[address]!.atBlock < self.RefillRequiredBlocks {
```

Comment:
This is not the safest approach, as someone can create thousands of accounts and repeatedly call this contract until the sender's balance turns to zero or less than the REFUEL amount.

Reply:
(discussed on call, but posting here for context) 
short term this is a known concern and we considered two short-term safety nets for this.
- The `REFUELER` account won't hold a large balance of FLOW 
- The `REFUELER` account will be siloed from our `PAYER` account. So if something like that does happen, `tryRefill()` can still be called with zero balance, and our `PAYER` account won't be effected. 


We are currently about to begin work on a `Dapper Identity Resource` to help identify Dapper users on-chain. This identity will only be granted by Dapper admins, so in the near future, we will have hooks to further ensure `DapperStorageRent` will only work with `Dapper Identities`, which users can't generate on mass.

cc @rrrkren